### DISCUSSION TITLE
experimentalJustInTimeCompile feedback

### DISCUSSION THREAD

**Discussion by AtofStryker** [cypress-io/cypress#30024] [Component Testing]
The `experimentalJustInTimeCompile` option was released in Cypress 13.14.0. Full details on its implementation can be found here: https://docs.cypress.io/guides/references/experiments#Experimental-Just-In-Time-Compile

We would like to move forward with making this default, but want to hear everyone's feedback on how this impacts their Component Tests, especially in terms of performance. 

Please leave feedback!

**Comment by mikeyavorsky** [comment]
I am seeing great improvements to performance, for a codebase with 132 component test specs. I'd be excited to make this default.

### Initial Compile
**Before**
```
468 assets
3282 modules
webpack 5.94.0 compiled successfully in 21129 ms
```
**After**
```
29 assets
862 modules
webpack 5.94.0 compiled successfully in 2999 ms
```

### Recompiling
**Before**
```
468 assets
3282 modules
webpack 5.94.0 compiled successfully in 22028 ms
```

**After**
```
33 assets
912 modules
webpack 5.94.0 compiled successfully in 2582 ms
```

**Comment by SergeyVolynkin** [comment]
@AtofStryker @jennifer-shehane Quick question if Cypress team evaluated migrating from Webpack to Turbopack/ESBuild/SWC for bundling? Adopting [Turbopack](https://turbo.build/) could potentially yield significant performance improvements, particularly in reducing startup times for both component and end-to-end tests.

**Comment by jasedwards** [comment]
I am running nx component tests for my angular components through an azure devops pipeline while connected to cypress cloud.  Before setting this value we would randomly get the chunk load error.  I don't know if this fixed it or not because I can't actually complete a run.  Because webpack now builds everything before each spec it errors out after 30 minutes.  I get a "disconnected" error which is fine as after 30 minutes it only ran 5 out of 60 specs.  Without the setting it is 30 minutes to run all specs.  I am searching for ways to make webpack cache in hoping that helps but have not found anything yet.  Ideally I could just use the same tools for cypress that I use for my apps (vite) and then I could move on from this problem.  Any suggestions how to run 1 spec in under 5 minutes with this setting would be greatly appreciated.

**Reply by AtofStryker** [reply]
@jasedwards since your app is built on vite, any reason you can't use `@cypress/vite-dev-server` and use `vite` as the `bundler` value?

**Reply by jasedwards** [reply]
Right now only webpack dev server is [supported ](https://docs.cypress.io/app/component-testing/get-started#Supported-Frameworks)for angular component tests.  To align with my app builds I would need cypress to support it and also for nx to remove the hard coding where it sets webpack as the bundler.  though I assume they will do that once other options are supported.

There was a [feature request](https://github.com/cypress-io/cypress/issues/30320) opened a few weeks ago but does not appear to be any acknowledgement of it yet.

**Reply by AtofStryker** [reply]
Ah I see. Do you have a sample reproduction we can look at or CI logs? What exactly do you mean by "Webpack now builds everything?"

A user in the chunk load error issue ran into something similar https://github.com/cypress-io/cypress/issues/28644#issuecomment-2382316793. Would you be able to try the patch to see how things respond? My guess is since you are also using angular the support file is also likely very large and might be causing problems

**Reply by jasedwards** [reply]
Sorry,  will give more detail on our setup.  For component testing instead of having a configuration for every nx library we just have one app called test-app.  this contains all of our component tests.  With this new setting webpack will rebuild test-app before every spec file (we have 60 right now).  This means before each spec there is a 5 minute wait before it can run.  I did figure out how to get caching working for webpack but that caused some test failures.  Honestly though if I could move to vite it still would not work with that setting because it even though vite only take 1 minute to build my apps it would still be 60 minutes of build time.

Regarding the patch, I was unclear on how to run with the patch if you could provide some detail on that I would appreciate it.

**Comment by HerrGustav** [comment]
@AtofStryker we still see these webpack chunk loading errors, even with the experimental jit compile feature enabled for component testing in Cypress v13.15.1. E.g. this one from today:

<img width="923" alt="image" src="https://github.com/user-attachments/assets/a560ff61-f89c-43db-9a51-395bca13f0a8">

It happens quite regularly and makes our ci really flaky. We use the default runners from Github actions, so they are not the most perfomant out there. But we do split our tests using cypress-split.

We use the normal webpack dev server provided by Cypress, configured for Next:
```
 devServer: {
      framework: 'next',
      bundler: 'webpack',
    },
``` 

Looking at the settings, the jit feature should be enabled: 
<img width="1100" alt="image" src="https://github.com/user-attachments/assets/fb8024dd-ddf6-48b2-b3b0-5b585c8251dc">

Our code base is not so large in this specific project, we are talking about 236 component test files split across 8 runners.
